modin/numpy/arr.py

```diff
@@ -1469,6 +1559,283 @@ def sum(
                 + initial
             )
 
+    def all(self, axis=None, out=None, keepdims=None, *, where=True):
+        check_kwargs(keepdims=keepdims, where=where)
+        if self._ndim == 1:
+            if axis == 1:
+                raise numpy.AxisError(1, 1)
+            result = self._query_compiler.all(axis=0)
+            if keepdims:
+                if out is not None and out.shape != (1,):
+                    raise ValueError(
+                        f"operand was set up as a reduction along axis 0, but the length of the axis is {out.shape[0]} (it has to be 1)"
+                    )
+                return fix_dtypes_and_determine_return(result, 1, bool, out, where)
+            result = result.to_numpy()[0, 0]
+            return result if not where else result and where
+        if axis is None:
+            result = self.flatten().all(
```

Comment:
We shouldn't be using flatten - it's pretty inefficient. We can set this up as two reductions - first along axis 1 then axis 0